sema/commons/j2/j2_functions.py

```
@@ -46,43 +52,77 @@
         content = asdbl
     # serialize to string again
     return xsd_value(str(content), quote, "xsd:double")
 
 
-def xsd_format_date(content, quote, suffix):
+def xsd_format_date(content: any, quote: str, *_) -> str:
     # make rigid date
+    if isinstance(content, datetime):
+        raise ValueError(
+            "use datetime format for datetime values, or past .date() result"
+        )
```

Ruff (0.8.2)

Prefer `TypeError` exception for invalid type

(TRY004)